src/main/scala/ai/privado/rulevalidator/YamlFileValidator.scala

```diff
@@ -0,0 +1,195 @@
+              ruleFile.contentAsString()
+            )
+          val schemaFile = matchSchemaFile(ruleFile, yamlAsJson, CommandConstants.VALIDATE)
+          if (schemaFile != null)
+            ValidationFailure(ruleFile.validateJsonFile(schemaFile, yamlAsJson), ruleFile)
+          else null
+        }
+      ).filter(vf => vf != null)
+
+    validationErrors
+      .filter(
+        ve =>
+          !ve.validationMessages.isEmpty
+      )
+  }
+
+  /**
+    * Validate a single rule file.
+    * Identify corresponding schema file for the input file and validate.
+    * @param ruleFile better.files.File object of the rule file to be validated
+    * @param configDirectory better.files.Files config location
+    * @return Boolean stating whether the rule file is valid
+    */
+  def isValidRuleFile(ruleFile: File, configDirectory: File): Boolean = {
+    if (!ruleFile.pathAsString.contains(s"${configDirectory.pathAsString}/$RULES_FOLDER_IN_CONFIG_DIR")) {
+      return false
+    }
+    val yamlAsJson = mapper
+      .readTree(
+        ruleFile.contentAsString()
+      )
+    val schemaFileContent = matchSchemaFile(ruleFile, yamlAsJson)
+    if (schemaFileContent == null)
+      false
+    else {
+      val validationMessages = ruleFile.validateJsonFile(schemaFileContent, yamlAsJson).asScala
+      if (validationMessages.nonEmpty) {
+        validationMessages.foreach(vm => {
+          println(s"File ${ruleFile.pathAsString} has following problems, ignoring file ...")
+          println(s"${vm.getMessage}")
+        })
+        false
+      }
+      else true
+    }
+  }
+
+  /**
+    * Find appropriate schema file to validate the rule file against it
+    * @param ruleFile better.files.File object for a YAML rule file to be validated
+    * @param ruleJsonTree com.fasterxml.jackson.databind.JsonNode object containing json data of rule file
+    *  @param callerCommand String value to govern pretty print of validation messages
+    * @return an Iterator on ai.privado.rulevalidator.ValidationFailure objects
+    */
+  def matchSchemaFile(ruleFile: File, ruleJsonTree: JsonNode, callerCommand: String = ""): String = {
+
+    val catLevelOneKey = if (ruleJsonTree.fieldNames().hasNext) ruleJsonTree.fieldNames().next() else CatLevelOne.UNKNOWN.name
+
+
+    CatLevelOne
+      .withNameWithDefault(
+        catLevelOneKey
+      ) match {
+      case CatLevelOne.SOURCES => SOURCES
+      case CatLevelOne.POLICIES => POLICIES
+      case CatLevelOne.THREATS => THREATS
+      case CatLevelOne.COLLECTIONS => COLLECTIONS
+      case CatLevelOne.SINKS => SINKS
+      case _ =>
+        if (callerCommand == CommandConstants.VALIDATE) println(PRETTY_LINE_SEPARATOR)
+        println(
+          f"File : ${ruleFile.pathAsString} :Adding new rules under the category '$catLevelOneKey'" +
+            f" is not supported. Ignoring file ...."
+        )
+//        if (callerCommand == CommandConstants.VALIDATE) println(PRETTY_LINE_SEPARATOR)
+        null
```

Comment:
Instead of returning `null` return `None` and above return `Some(SOURCES)` etc, usage of null should be prevented